### PR TITLE
Cleanup

### LINKED ISSUE #10: You cannot delete a question in the same session you added it.

### LINKED ISSUE #11: Create UserService for frontend to interact with backend

### PR DESCRIPTION
**Pull request by igezt**
- Cleanup frontend and backend file system.
- Fixed bug #10
- Added a loading spinner for questions,
- Implemented #11

### CODE REVIEW

File: apps/backend/src/database/prisma.ts [hunk 1]
```diff
@@ -1,34 +1,3 @@
 import { PrismaClient } from "@prisma/client";
 
 export const prisma = new PrismaClient();
-// Example usage: This find the first user that has the email "abc@gmail.com" with a username 'alksdja"
```
Comment (Douglch) [inline apps/backend/src/database/prisma.ts:4]: oops, thanks for removing this

File: apps/frontend/.env [hunk 2]
```diff
@@ -0,0 +1,2 @@
+BASE_URL=http://localhost:8000
```
Comment (Douglch) [inline apps/frontend/.env:1]: Hol up, is this supposed to be here

**Review by Douglch — APPROVED** [review]
Thanks for refactoring and cleaning up the code :shipit:

File: apps/frontend/src/components/questions/AddQuestionModal.tsx [hunk 3]
```diff
@@ -1,36 +1,45 @@
...
+    value,
+    valueIsValid,
+    hasError,
+    valueChangeHandler,
+    inputBlurHandler,
+    reset,
+  } = useInput(
+    (s: string) =>
+      s.trim().length > 0 &&
+      questions.filter((question: Question) => question.title == s).length != 1
```
Comment (huanghao1998) [inline apps/frontend/src/components/questions/AddQuestionModal.tsx:33]: This should be 
```
questions.filter((question: Question) => question.title == s).length == 0
```

File: apps/frontend/src/components/questions/AddQuestionModal.tsx [hunk 4]
```diff
@@ -79,24 +88,35 @@ const AddQuestionModal: React.FC<AddQuestionModalProps> = ({
...
-                ? <label className="error-message" style={{ color: 'red' }}>Question already exists.</label>
-                : null}
+                  {hasError && value.trim().length <= 0 ? (
+                    <label className="error-message" style={{ color: "red" }}>
+                      Input cannot be empty.
+                    </label>
+                  ) : hasError &&
+                    questions.filter(
+                      (question: Question) => question.title == value
+                    ).length == 1 ? (
```
Comment (huanghao1998) [inline apps/frontend/src/components/questions/AddQuestionModal.tsx:115]: this should be
```
 .length > 0
```

**Review by huanghao1998 — COMMENTED** [review]
Just change those it would work

**Review by Jarrett0203 — APPROVED** [review]

File: apps/frontend/.env (hunk 2 above)
Comment (Guo-KeCheng) [reply apps/frontend/.env:1]: dont think see eh @igezt did you accidentally added this

File: apps/backend/prisma/schema.prisma [hunk 5]
```diff
@@ -17,4 +17,24 @@ model User {
...
+  description String
+  categories Category[]
+  complexity Complexity
+}
+
+model Category {
+  id Int @id @default(autoincrement())
+  name String
+  questions Question[]
+}
```
Comment (Guo-KeCheng) [inline apps/backend/prisma/schema.prisma:34]: Can I just check for this part I thought is handled by MongoDB or psql side also handling questions? @igezt

**Review by Guo-KeCheng — CHANGES_REQUESTED** [review]

File: apps/frontend/.env (hunk 2 above)
Comment (igezt) [reply apps/frontend/.env:1]: HAHHAH it was here from the dev branch. I think somebody added it in last time. Ill remove it 😆

File: apps/backend/prisma/schema.prisma (hunk 5 above)
Comment (igezt) [reply apps/backend/prisma/schema.prisma:34]: This is just for the future when we move over to psql

**Comment by igezt** [comment]
Ill resolve the issues in a new branch after merging this branch and Jarret's branch in because I have changes to make but Jarret already merged in my branch into his.

**Pull request merged**